pkg/workflow/tekton/response.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tekton
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	pb "sqlflow.org/sqlflow/pkg/proto"
+)
+
+// CompoundResponses Compounds response
+type CompoundResponses struct {
+	stepCnt   int
+	stepIdx   int
+	responses []*pb.Response
+}
+
+// NewCompoundResponsesWithStepIdx returns CompoundResponses with step index
+func NewCompoundResponsesWithStepIdx(stepCnt, stepIdx int) *CompoundResponses {
+	return &CompoundResponses{
+		stepCnt:   stepCnt,
+		stepIdx:   stepIdx,
+		responses: []*pb.Response{},
+	}
+}
+
+// NewCompoundResponses returns CompoundResponses with step index 0
+func NewCompoundResponses() *CompoundResponses {
+	return &CompoundResponses{
+		stepCnt:   0,
+		stepIdx:   0,
+		responses: []*pb.Response{},
+	}
+}
+
+// AppendMessageWithStepIdx append message with Step Index as prefix
+func (r *CompoundResponses) AppendMessageWithStepIdx(message string) error {
```

Comment:
`AppendMessageWithStepIdx` does not have `StepIdx` argument, so the caller is not aware of the `StepIdx`, can rename to `AppendMessage`?